### PR TITLE
Enable to extends module `card_list.html.twig` and use custom actions

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Enable to extends module list block actions. Add actions block in modules card_list.html.twig
| Type?             | improvement 
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| How to test?      | N/A
| Possible impacts? | N/A

## Ability to extends `card_list.html.twig` and use custom actions 

My custom module card list :

```twig
{% extends '@PrestaShop/Admin/Module/Includes/card_list.html.twig' %}
{% block module_actions %}
     // My custom action buttons here
{% endblock %}

```

image

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/23863)

### CODE REVIEW

**Review by NeOMakinG — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by sarahdib** [comment]
Hello @PrestaShop/product-team 

Could you check this improvement before QA ?

Thank you

**Comment by marionf** [comment]
Hello @sarahdib @PululuK 

Could you please create an issue to explain what is the bug / need and what is the solution you suggest to fix it ?

**Comment by PululuK** [comment]
> Hello @sarahdib @PululuK
> 
> Could you please create an issue to explain what is the bug / need and what is the solution you suggest to fix it ?

Is not a bug, is purely a dev feature ! So I think no need issue related to.

**Comment by marionf** [comment]
Well, if it's a dev feature without impact in the BO or functionnal behavior, the feature should be approved by a maintainer not a PM :)

**Review by matthieu-rolland — APPROVED** [review]

**Review by sowbiba — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by Robin-Fischer-PS** [comment]
Hello @PululuK !

Can you provide a How to Test, for QA ? If your feature is too technical to test, it will be tested by another dev :smile: 

Thanks !

**Comment by PululuK** [comment]
> Hello @PululuK !
> 
> Can you provide a How to Test, for QA ? If your feature is too technical to test, it will be tested by another dev smile
> 
> Thanks !

Hi @Robin-Fischer-PS 
Yes is a technical feature, @NeOMakinG can you please test it ?

## How to test 

Basically  

- Edit  `src/PrestaShopBundle/Resources/views/Admin/Module/Includes/card_manage_installed.html.twig` and add the follow code (don't remove the old code)
```twig 
{% block module_actions %}
    <button class='btn btn-danger-reverse btn-outline-danger' href='#' >
        <i class="material-icons">delete</i> My Custom button
    </button>
{% endblock %}
```
- go to BO > Modules manager and see you have 


image

**Comment by NeOMakinG** [comment]
image

It works, but do we really want to accept to remove basical actions? A module could avoid to disable any modules and that could be an issue don't you think?

**Comment by PululuK** [comment]
> It works, but do we really want to accept to remove basical actions? A module could avoid to disable any modules and that could be an issue don't you think?

I do not understand what you mean @NeOMakinG 😀
This PR is about `extends` and use custom actions, so it is up to the developer to decide whether or not he wants to delete the default actions and use these own actions.

**Comment by NeOMakinG** [comment]
> > It works, but do we really want to accept to remove basical actions? A module could avoid to disable any modules and that could be an issue don't you think?
> 
> I do not understand what you mean @NeOMakinG 😀
> This PR is about `extends` and use custom actions, so it is up to the developer to decide whether or not he wants to delete the default actions and use these own actions.

I mean, you could kick the uninstall action, and a normal user won't be able to uninstall the module for example?

**Comment by Progi1984** [comment]
@NeOMakinG May be it should be decided by PM. Ping @PrestaShop/product-team

**Comment by PululuK** [comment]
Hi @NeOMakinG @Progi1984 @PrestaShop/prestashop-core-developers 

I'm not sure we're talking about the same thing or understanding PR the same 😄

> I mean, you could kick the uninstall action, and a normal user won't be able to uninstall the module for example?

```html
Template Inheritance

Inheritance brings the concept of Object Oriented Programming to templates, 
allowing you to define one (or more) base templates that can be extended by child templates. 
Extending means that the child template can override all or some of the parent named block areas. 
```
https://www.smarty.net/docs/en/advanced.features.template.inheritance.tpl

so it's up to the developer to decide what he wants in his child template.

Ex.: 

- this is my custom module card list (extends the core car list)
- as this is a custom listing, I must have the option to display my own buttons if I want (it is the interest of my PR).
- basically "c'est mon template, je fais ce que je veux !"  😀

image

**Comment by PierreRambaud** [comment]
I validate it as it's the right behavior, he only adds a Twig block and it helps to extend templates.

**Comment by PierreRambaud** [comment]
Thank you @PululuK
